File: leetcode/golang/problems/429. N 叉树的层序遍历.go
package problems

/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

 func levelOrder(root *Node) (ans [][]int) {
    q := [](*Node){}
	q = append(q, root)
	ans = make([][]int, 0)
	ans = append(ans, []int{root.Val})
	for len(q) > 0 {
		size := len(q)
		path := []int{}
		for i := 0; i < size; i++ {
			node := q[0]
			q = q[1:]
			for j := 0; j < len(node.Children); j++ {
				q = append(q, node.Children[j])
				path = append(path, node.Children[j].Val)
			}
		}
        temp := make([]int, len(path))
        copy(temp, path)
        ans = append(ans, temp)
	}
	return
}

File: leetcode/cpp/429. N 叉树的层序遍历.cpp
#include <bits/stdc++.h>
#include <queue>
#include <vector>

using namespace std;

class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};

vector<vector<int>> levelOrder(Node* root) {
    queue<Node*> q;
    q.push(root);
    vector<vector<int>> ans;
    ans.push_back(vector<int>({root->val}));
    while (q.size() > 0) {
        int size = q.size();
        while (size--) {
            Node* node = q.front();
            q.pop();
            vector<int> path;
            for (int i = 0; i < node->children.size(); i++) {
                q.push(node->children[i]);
                path.push_back(node->children[i]->val);
            }
            ans.push_back(path);
        }
    }
    return ans;
}

int main() {

}